src/solver/optimisation/opt_optimisation_lineaire.cpp

```diff
@@ -158,14 +160,42 @@ void runThermalHeuristic(PROBLEME_HEBDO* problemeHebdo)
         OPT_CalculerLesPminThermiquesEnFonctionDeMUTetMDT(problemeHebdo);
     }
 }
+
+
+void resizeProbleme(PROBLEME_ANTARES_A_RESOUDRE* ProblemeAResoudre,
+                    unsigned nombreDeVariables,
+                    unsigned nombreDeContraintes)
+{
+    ProblemeAResoudre->CoutQuadratique.resize(nombreDeVariables);
+    ProblemeAResoudre->CoutLineaire.resize(nombreDeVariables);
+    ProblemeAResoudre->TypeDeVariable.resize(nombreDeVariables);
+    ProblemeAResoudre->Xmin.resize(nombreDeVariables);
+    ProblemeAResoudre->Xmax.resize(nombreDeVariables);
+    ProblemeAResoudre->X.resize(nombreDeVariables);
+    ProblemeAResoudre->AdresseOuPlacerLaValeurDesVariablesOptimisees.resize(nombreDeVariables);
+    ProblemeAResoudre->AdresseOuPlacerLaValeurDesCoutsReduits.resize(nombreDeVariables);
+    ProblemeAResoudre->PositionDeLaVariable.resize(nombreDeVariables);
+    ProblemeAResoudre->Pi.resize(nombreDeVariables);
+    ProblemeAResoudre->Colonne.resize(nombreDeVariables);
```

Comment:
All is good except for this. These are temporary variables used in `ConstraintBuilder::OPT_ChargerLaContrainteDansLaMatriceDesContraintes`. You don't need to resize them